packages/core/node/core-node-server-internal/src/node_config.ts

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { schema, TypeOf } from '@kbn/config-schema';
+import type { ServiceConfigDescriptor } from '@kbn/core-base-server-internal';
+
+/** @internal */
+export const NODE_CONFIG_PATH = 'node' as const;
+/** @internal */
+export const NODE_WILDCARD_CHAR = '*';
+/** @internal */
+export const NODE_ACCEPTED_ROLES = ['background-tasks', 'ui'];
+
+const configSchema = schema.object({
+  roles: schema.oneOf(
+    [
+      schema.arrayOf(schema.oneOf([schema.literal('background-tasks'), schema.literal('ui')])),
```

Comment:
Have we considered matching casing with elasticsearch? https://www.elastic.co/guide/en/elasticsearch/reference/current/modules-node.html#node-roles

e.g. `s/background-tasks/background_tasks`

Reply:
That's a fair point, @jbudz. FWIW, when I was writing the RFC, I didn't intentionally deviate from the way that ES does this, it was just what my fingers typed by default. I'm supportive of either approach.